mito-sql-cell/src/sqltoolbar.tsx

```diff
@@ -0,0 +1,289 @@
+  private _variableName = '';
+
+  /**
+   * The database name.
+   */
+  get database(): string {
+    return this._database;
+  }
+  set database(v: string) {
+    if (v !== this._database) {
+      this._database = v;
+      this.stateChanged.emit();
+    }
+  }
+
+  /**
+   * The SQL query result variable name.
+   */
+  get variableName(): string {
+    return this._variableName;
+  }
+  set variableName(v: string) {
+    if (v !== this._variableName) {
+      this._variableName = v;
+      this.stateChanged.emit();
+    }
+  }
+}
+
+/**
+ * SQL toolbar constructor argument
+ */
+export interface ISQLToolbarOptions {
+  /**
+   * Command registry
+   */
+  commands: CommandRegistry;
+  /**
+   * Cell model
+   */
+  model: ICellModel;
+  /**
+   * SQL sources
+   */
+  sqlSources: ISqlSources;
+}
+
+/**
+ * SQL toolbar widget.
+ */
+export class SQLToolbar extends VDomRenderer<SqlModel | null> {
+  private _cellModel: ICellModel;
+  private _commands: CommandRegistry;
+  private _sqlSources: ISqlSources;
+
+  /**
+   * Creates a cell header.
+   */
+  constructor(options: ISQLToolbarOptions) {
+    super();
+    this._cellModel = options.model;
+    this._commands = options.commands;
+    this._sqlSources = options.sqlSources;
+    this.addClass(TOOLBAR_CLASS);
+
+    this.modelChanged.connect(this._onModelChanged, this);
+    this._parseSource();
+
+    this._cellModel.sharedModel.changed.connect(
+      this._onSharedModelChanged,
+      this
+    );
+    this._sqlSources.changed.connect(this._onSQLSourcesChanged, this);
+  }
+
+  /**
+   * Dispose the widget and its model.
+   */
+  dispose(): void {
+    if (this.isDisposed) {
+      return;
+    }
+    this.model?.dispose();
+    super.dispose();
+  }
+
+  protected render(): JSX.Element | null {
+    return this.model ? (
+      <Toolbar aria-label="Cell SQL toolbar">
+        <Select
+          title="SQL source"
+          onChange={this._onDatabaseChange}
+          scale="xsmall"
+          value={this.model.database}
+        >
+          <Option
+            key="add"
+            className="mito-sql-add-option"
+            value={ADD_SOURCE_OPTION_VALUE}
+            onClick={this._addDatabase}
+          >
+            <addIcon.react tag={null} slot="start" />
+            Connect to a database…
+          </Option>
+          {map(this._sqlSources, s => (
+            <Option key={s.connectionName} value={s.connectionName}>
+              {s.connectionName}
+            </Option>
+          ))}
+        </Select>
+        <span style={{ margin: 'auto var(--toolbar-item-gap)' }}>saved to</span>
+        <TextField
+          aria-label="Variable name"
+          title="Variable name"
+          onInput={this._onVariableChange}
+          onChange={this._onVariableChange}
+          placeholder="Variable name"
+          value={this.model.variableName}
+        >
+          <editIcon.react slot="end" tag={null} />
+        </TextField>
+      </Toolbar>
+    ) : null;
```

Comment:
Is it possible to display some hint text for each database type once the user select it? I think since there are a bunch of different ways to connect to a database and different connections will have different optional fields filled out, users are going to have a hard time knowing exactly the right way to write their query for this particular connection approach.

Reply:
I guess this is probably a feature we should leave for v2 though.